app/submitter/converter.py

```diff
@@ -82,14 +83,30 @@ def convert_answers_to_data(answer_store, questionnaire_schema):
     for answer in answer_store.answers:
         item = questionnaire_schema.get_item_by_id(answer['answer_id'])
         value = answer['value']
+
         if item is not None and value is not None:
-            if item.code not in data:
-                data[item.code] = _encode_value(value)
+            if not isinstance(item, CheckboxAnswer) or any('q_code' not in option for option in item.options):
+
+                if item.code not in data:
+                    data[item.code] = _encode_value(value)
+                else:
+                    if not isinstance(data[item.code], list):
+                        list_answers = [data[item.code]]
+                        data[item.code] = list_answers
+                    data[item.code].append(_encode_value(value))
             else:
-                if not isinstance(data[item.code], list):
-                    list_answers = [data[item.code]]
-                    data[item.code] = list_answers
-                data[item.code].append(_encode_value(value))
+                for user_answer in value:
```

Comment:
Similarly, I think this whole for loop would be better broken into it's own method which returns checkbox data for a item.code. Like _get_checkbox_answer_data?

Breaking them both out should make it very clear what's going on here.